org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java

```diff
@@ -703,4 +709,44 @@ public void projectsBuildFinished(IProgressMonitor monitor) {
 	public boolean isBuildFinished() {
 		return buildFinished;
 	}
+
+	@Override
+	public void checkIndexes() {
+		Job job = new Job(("Check Indexes")) {
+
+			@Override
+			protected IStatus run(IProgressMonitor monitor) {
+				IJavaProject[] javaProjects = ProjectUtils.getJavaProjects();
+				if (javaProjects == null || javaProjects.length <= 0) {
```

Comment:
Could `javaProjects` ever be `null` ? Seems like it will always return at worst an empty array (looking into how its implemented), in which case `javaProjects.length == 0` should be enough.

Reply:
Fixed.